uint16 class_name
uint16 x0
uint16 y0
uint16 x1
uint16 y1